slides/02.qmd

```diff
@@ -5,57 +5,32 @@ title: Names and values
 
 ## Learning objectives
 
-- To be able to understand distinction between an *object* and its *name*
-- With this knowledge, to be able write faster code using less memory
+- To distinction between an *object* and its *name*
+- To identify when data are _physically_ copied versus modified
 - To better understand R's functional programming tools
 
-Using lobstr package here.
-```{r}
+The lobstr package will help us through the chapter
+
+```{r, echo = TRUE}
 library(lobstr)
 ```
 
 
 ## Quiz
 
-### 1. How do I create a new column called `3` that contains the sum of `1` and `2`?
+**1. Syntactic vs Non-syntactic names?**
 
-```{r}
-df <- data.frame(runif(3), runif(3))
-names(df) <- c(1, 2)
-df
-```
+Syntactic names: contain alphanumeric digits, `.`, and  `_`
 
-```{r}
-df$`3` <- df$`1` + df$`2`
-df
-```
-
-**What makes these names challenging?**
-
-> You need to use backticks (`) when the name of an object doesn't start with a 
-> a character or '.' [or . followed by a number] (non-syntactic names).
-
-### 2. How much memory does `y` occupy?
-
-```{r}
-x <- runif(1e6)
-y <- list(x, x, x)
-```
+Non-syntactic names: Start with a `_` or 0-9. Workaround is to surround in ```.
 
-Need to use the lobstr package:
-```{r}
-lobstr::obj_size(y)
-```
+Also, can't use reserved words found at `?Reserved`. 
 
-> Note that if you look in the RStudio Environment or use R base `object.size()`
-> you actually get a value of 24 MB
+## Quiz
 
-```{r}
-object.size(y)
-```
+**2. On which line does `a` get copied in the following example?**
 
-### 3. On which line does `a` get copied in the following example?
-```{r}
+```{r, echo = TRUE}
```

Comment:
```suggestion
```{r}
```
Our defaults are TRUE for echo and eval (see `_metadata.yml`)

Reply:
For some reason it wasn't echo-ing when I previewed so I added them. But my mistake for not removing them! Will do in the update

Reply:
It MIGHT depend how you preview. I'm guessing there's a way to render *just* this one file, and that probably won't pick up the site settings. Ctrl-shift-k on this file or `quarto render` on the site as a whole should render it properly.